packages/p2p/src/stores/general-store.js

```diff
@@ -638,6 +641,10 @@ export default class GeneralStore extends BaseStore {
         this.contact_info = contact_info;
     }
 
+    setCounterpartyAdvertId(counterparty_advert_id) {
+        this.counterparty_advert_id = counterparty_advert_id;
+    }
+
```

Comment:
Can this be handled in the component? Let's not add more logic to stores.

Reply:
ideally it would be best, let me try and see if we can move it to the component buttt i added this because i needed to track if there was an advert id, and when the user navigates to the shared ad, the advert_id in search param disappears and reappears, so it kept causing issues when trying to show the advert... hence why i needed this observable... it was a similar issue for the advertiser id and the order id when niloofar and I changed the routing

Reply:
so i tried to handle everything in app.jsx where i used useState to track the advert_id in search params. This did work, but the modal couldn't close because it still had a value within the state. And because that state is only handled in the app, i can't change the value... You can see that i had set `counterparty_advert_id' to an empty string so it wont show the modal. 

i did try to reset the state in app, but then the modal wouldn't show